src/pages/chat.tsx

```diff
@@ -1,18 +1,49 @@
+//@ts-nocheck
 import type { NextPage } from 'next';
 import Image from 'next/image';
 import avatarMale from '../images/avatar/male.png';
 import Chat from '../components/chat';
-import ChatBox from '../components/chatbox';
+import { useState, useRef } from 'react';
+import { useSession, signOut } from 'next-auth/react';
+
+// Temporarily store messages of a user
+const messages = [];
 
 const ChatPage: NextPage = (props): JSX.Element => {
+    const { data: session } = useSession();
+    const dummy = useRef();
+    const [chatText, setChatText] = useState('');
+
+    if (!session) return <p>Access Denied.</p>;
+
+    const userTyping = (event: any) => {
+        // keycode for when the user presses the Enter button
+        event.keyCode === 13
+            ? submitMessage()
+            : setChatText(event.target.value);
+    };
+
+    // Check if message is empty or just blank spaces
+    const messageValid = (txt: any) => txt && txt.replace(/\s/g, '').length;
+
+    // Submit the message if it is valid
+    const submitMessage = () => {
+        if (messageValid(chatText)) {
+            document.getElementById('chatInput').value = '';
+            messages.push(chatText);
```

Comment:
This is fine for now but once you read the socket server PR for matching, the socket server will actually emit every message it receives to every client in the room. That means that the client will also receive his/her own message, which we will render as a chat component.